src/pushNotifications/widget/pushNotifications.js

```diff
@@ -297,17 +297,41 @@ define([
                 deviceRegistration.set("DeviceType", "iOS");
             }
 
-            // We commit the DeviceRegistration object to trigger the backend process.
-            // This is the only time we can commit the object, because it will be deleted in the AfterCommit event.
-            mx.data.commit({
-                mxobj: deviceRegistration,
-                callback: dojoLang.hitch(this, function() {
-                    logger.debug("Registered device with ID " + deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
-                }),
-                error: function(e) {
-                    logger.error("Error occurred attempting to register device: " + e);
-                }
-            });
+            // CC Jan 10, 2018 - only commit the DeviceRegistration if the registration id has changed or does not yet exist
+            var lastRegistrationId = localStorage.getItem("mx-registration-id");
+            if (!lastRegistrationId || lastRegistrationId !== deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE)) {
+                // We commit the DeviceRegistration object to trigger the backend process.
+                // This is the only time we can commit the object, because it will be deleted in the AfterCommit event.
+                mx.data.commit({
+                    mxobj: deviceRegistration,
+                    callback: dojoLang.hitch(this, function() {
+                        logger.debug("Registered device with ID " + deviceRegistration.get(this.REGISTRATION_ID_ATTRIBUTE));
+                        if (mx.isOffline()) {
+                            mx.data.synchronizeOffline({ // settings, success, error
```

Comment:
Is this part of the public API?

Reply:
is what part of the public api? `synchronizeOffline`? doesn't look like it 😬

Reply:
It isn't, indeed :P I asked Danny, and he said it's so often used that it's sort of ok to use it though...